packages/core/src/providers/pinpoint/apis/updateEndpoint.ts

```diff
@@ -46,22 +47,34 @@ export const updateEndpoint = async ({
 		name,
 		plan,
 	} = userProfile ?? {};
-	const clientInfo = getClientInfo();
+
+	// only automatically populate the endpoint with client info and identity id upon endpoint creation to
+	// avoid overwriting the endpoint with these values every time the endpoint is updated
+	const demographicsFromClientInfo: UserProfile['demographic'] = {};
+	const resolvedUserId = createdEndpointId ? userId ?? identityId : userId;
```

Comment:
Nit: Did we decide that `||` is better to use than `??`
```suggestion
	const resolvedUserId = createdEndpointId ? userId || identityId : userId;
```

Reply:
To make sure I understand: the `identityId` property is always a default value and `userId` is always customer provided or empty, correct?

Reply:
If this call creates an endpoint
* use `userId` if provided, otherwise fall back to `identityId`

If this call updates an endpoint
* use `userId` if provided but don't fall back to anything